charset_normalizer/models.py

```diff
@@ -225,12 +225,10 @@ def has_submatch(self) -> bool:
     def alphabets(self) -> List[str]:
         if self._unicode_ranges is not None:
             return self._unicode_ranges
-        detected_ranges = set()  # type: Set[str]
-        for character in str(self):
-            detected_range = unicode_range(character)  # type: Optional[str]
-            if detected_range:
-                detected_ranges.add(detected_range)
-        self._unicode_ranges = sorted(list(detected_ranges))
+        # list detected ranges
+        detected_ranges = [unicode_range(char) for char in str(self)]  # List[str]
+        # filter and sort
+        self._unicode_ranges = sorted([r for r in detected_ranges if r is not None])  # List[str]
```

Comment:
This has different semantics c.f. the original. This will not drop any empty ranges (if there are any) unlike the original did.

Reply:
Why not, but I don't think it changes anything. Otherwise the tests wouldn't pass, don't you think?

Reply:
Well, if you're doing changes for performance that shouldn't change semantics...

Either way, it's a tiny bit faster:

```
~ $ python3 -m timeit -s 'a = ["u","o",None,""]' 'sorted([v for v in a if v is not None])'
1000000 loops, best of 5: 367 nsec per loop
~ $ python3 -m timeit -s 'a = ["u","o",None,""]' 'sorted([v for v in a if v])'
1000000 loops, best of 5: 334 nsec per loop
```

Reply:
OK, fair enough.

Reply:
Yes. Absolutely.